### ISSUE TITLE
Unable to target gateway deployment due to generated name

### ISSUE THREAD

**Issue by marcoferrer** [kumahq/kuma#3938]
### What happened?

When defining a gateway the resulting deployment uses a generated name as configured here
https://github.com/kumahq/kuma/blob/061f373181eec996d51c0cd2fba62a6986e78f71/pkg/plugins/runtime/k8s/controllers/gateway_instance_controller.go#L176

The problem that Im facing now is that since the name is generated I cant target this deployment with an Hpa or Pdb. Would it make sense to have the deployment name match the gateway instance name? Considering that the deployment is created in the same namespace as the gateway instance. Or is there a particular reason why this is need that not immediately clear?

**Comment by lahabana** [comment]
Makes sense we should indeed switch it to be deterministic and you the same name.
